src/core/utils/components/VirtualScroll.tsx

```diff
@@ -100,145 +132,224 @@ export type VirtualScrollProps = {
-    if (!onScrollRef.current) {
-      return;
-    }
-    !scrollContainer.current ||
-    scrollContainer.current === parentRef.current?.ownerDocument.body
-      ? parentRef.current?.ownerDocument.removeEventListener(
-          'scroll',
-          onScrollRef.current,
-        )
-      : scrollContainer.current.removeEventListener(
-          'scroll',
-          onScrollRef.current,
-        );
-  }, []);
-
-  // Add event listener to the scrollable container.
-  React.useLayoutEffect(() => {
-    removeScrollListener();
-    onScrollRef.current = updateVirtualScroll;
-    if (
+      if (!parentRef.current) {
+        return;
+      }
+      parentRef.current.style.transform = `translateY(${getTranslateValue(
+        childHeight.current.child,
+        startIndex,
+      )}px)`;
+    }, [bufferSize, itemsLength]);
+
+    const onScroll = React.useCallback(() => {
+      updateVirtualScroll();
+    }, [updateVirtualScroll]);
+
+    const removeScrollListener = React.useCallback(() => {
+      if (!onScrollRef.current) {
+        return;
+      }
       !scrollContainer.current ||
       scrollContainer.current === parentRef.current?.ownerDocument.body
-    ) {
-      parentRef.current?.ownerDocument.addEventListener(
-        'scroll',
-        updateVirtualScroll,
-      );
-    } else {
-      scrollContainer.current.addEventListener('scroll', updateVirtualScroll);
-    }
-    return removeScrollListener;
-  }, [updateVirtualScroll, removeScrollListener]);
-
-  React.useLayoutEffect(() => {
-    updateVirtualScroll();
-  }, [scrollContainerHeight, itemsLength, updateVirtualScroll]);
-
-  return (
-    <div
-      style={{
-        overflow: 'hidden',
-        minHeight: itemsLength * childHeight.current,
-        width: '100%',
-        ...style,
-      }}
-      ref={ref}
-      {...rest}
-    >
+        ? parentRef.current?.ownerDocument.removeEventListener(
+            'scroll',
+            onScrollRef.current,
+          )
+        : scrollContainer.current.removeEventListener(
+            'scroll',
+            onScrollRef.current,
+          );
+    }, []);
+
+    // Add event listener to the scrollable container.
+    React.useLayoutEffect(() => {
+      removeScrollListener();
+      onScrollRef.current = onScroll;
+      if (
+        !scrollContainer.current ||
+        scrollContainer.current === parentRef.current?.ownerDocument.body
+      ) {
+        parentRef.current?.ownerDocument.addEventListener('scroll', onScroll);
+      } else {
+        scrollContainer.current.addEventListener('scroll', onScroll);
+      }
+      return removeScrollListener;
+    }, [onScroll, removeScrollListener]);
+
+    React.useLayoutEffect(() => {
+      if (!scrollContainerHeight) {
+        return;
+      }
+
+      const scrollableContainer =
+        scrollContainer.current ??
+        (parentRef.current?.ownerDocument.scrollingElement as HTMLElement);
+
+      // if `scrollToIndex` is not visible, scroll to it
+      if (
```

Comment:
I think `scrollToIndex` doesn't need default value, it should be `undefined` if not set. Because it is scrolling to 0 if nothing is passed. Open virtualized Table story, scroll a bit down and resize the window -> table will scroll to the first row.

Reply:
Now I am thinking maybe there is a problem with passed `scrollToIndex` after you scroll down and resize, it should scroll to the previous index.

Reply:
left as undefined, if nothing is passed. It fixes table bug with resize...
2nd part will need to see, how to fix

Reply:
Fixed 2nd part...